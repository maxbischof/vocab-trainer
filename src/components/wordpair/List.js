import React from 'react'
import styled from 'styled-components'
import WordPair from './WordPair'

export default function List({ words, deleteWord }) {
  return (
    <StyledWordPairList>
      {words.map((word, index) => (
        <WordPair
          word={word}
          index={index}
          deleteWord={deleteWord}
          key={word.foreign}
        />
      ))}
    </StyledWordPairList>
  )
}

const StyledWordPairList = styled.ul`
  list-style-type: none;
  padding: 0;
  display: grid;
  justify-items: center;
`
