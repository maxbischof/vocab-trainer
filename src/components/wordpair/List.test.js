import React from 'react'
import List from './List'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const words = [
    { foreign: 'le ordinateur', native: 'der Computer' },
    { foreign: 'la Ã©cole', native: 'die Schule' },
  ]
  const tree = renderer.create(<List words={words} />).toJSON()
  expect(tree).toMatchSnapshot()
})
