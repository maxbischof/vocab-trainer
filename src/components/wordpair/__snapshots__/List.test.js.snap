// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="sc-hKgILt gAqaJR"
>
  <li
    className="sc-bdfBwQ ldQlgV"
  >
    <div
      className="sc-gsTCUz eugYRC"
    >
      <b>
        le ordinateur
      </b>
      <small>
        der Computer
      </small>
    </div>
    <svg
      className="sc-dlfnbm lnKTpL"
      onClick={[Function]}
    >
      delete.svg
    </svg>
  </li>
  <li
    className="sc-bdfBwQ ldQlgV"
  >
    <div
      className="sc-gsTCUz eugYRC"
    >
      <b>
        la Ã©cole
      </b>
      <small>
        die Schule
      </small>
    </div>
    <svg
      className="sc-dlfnbm lnKTpL"
      onClick={[Function]}
    >
      delete.svg
    </svg>
  </li>
</ul>
`;
